cpp/include/cuml/linear_model/glm.hpp

```diff
@@ -226,177 +177,93 @@ void qnFit(const raft::handle_t& cuml_handle,
+                 T* f,
                  int* num_iters,
-                 int loss_type,
-                 double* sample_weight = nullptr);
-/** @} */
+                 T* sample_weight = nullptr);
 
 /**
- * @defgroup qnDecisionFunction to obtain the confidence scores of samples
+ * @brief Obtain the confidence scores of samples
+ *
  * @param cuml_handle           reference to raft::handle_t object
- * @param X                     device pointer to feature matrix of dimension NxD (row- or column
- * major: see X_col_major param)
+ * @param pams                  model parameters
+ * @param X                     device pointer to feature matrix of dimension NxD (row- or
+ * column-major: see X_col_major param)
  * @param X_col_major           true if X is stored column-major, i.e. feature columns are
  * contiguous
  * @param N                     number of examples
  * @param D                     number of features
  * @param C                     number of outputs (C > 1, for multinomial, indicating number of
  * classes. For logistic, C = 2 and normal, C = 1.)
- * @param fit_intercept         true if model includes a bias.
  * @param params                device pointer to model parameters. Length D if fit_intercept ==
  * false else D+1
- * @param loss_type             id of likelihood model (0: logistic/sigmoid, 1: multinomial/softmax,
- * 2: normal/squared)
  * @param scores                device pointer to confidence scores of length N (for binary
  * logistic: [0,1], for multinomial:  [0,...,C-1])
- * @{
  */
+template <typename T>
 void qnDecisionFunction(const raft::handle_t& cuml_handle,
-                        float* X,
-                        bool X_col_major,
-                        int N,
-                        int D,
-                        int C,
-                        bool fit_intercept,
-                        float* params,
-                        int loss_type,
-                        float* scores);
-void qnDecisionFunction(const raft::handle_t& cuml_handle,
-                        double* X,
+                        const qn_params& pams,
+                        T* X,
                         bool X_col_major,
                         int N,
                         int D,
                         int C,
-                        bool fit_intercept,
-                        double* params,
-                        int loss_type,
-                        double* scores);
-/** @} */
+                        T* params,
+                        T* scores);
 
 /**
- * @defgroup qnDecisionFunctionSparse to obtain the confidence scores of samples
+ * @brief Obtain the confidence scores of samples
+ *
  * @param cuml_handle           reference to raft::handle_t object
- * @param X_values              feature matrix values (CSR format),
- * matrix dimension: NxD.
+ * @param pams                  model parameters
+ * @param X_values              feature matrix values (CSR format), matrix dimension: NxD.
```

Comment:
`X_values` and `X_cols` should be vectors of length `X_nnz`. `X_cols` range should be (0, D). `X_row_ids` should be a vector of length `N` + 1.